cpp/examples/advanced/cugraph_operations/graph_operations.cu

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../tests/utilities/base_fixture.hpp"
+#include "../tests/utilities/test_utilities.hpp"
+
+#include <prims/per_v_transform_reduce_incoming_outgoing_e.cuh>
+#include <prims/reduce_op.cuh>
+#include <prims/update_edge_src_dst_property.cuh>
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/edge_partition_view.hpp>
+#include <cugraph/edge_src_dst_property.hpp>
+#include <cugraph/graph_view.hpp>
+
+#include <raft/comms/mpi_comms.hpp>
+#include <raft/core/comms.hpp>
+#include <raft/core/handle.hpp>
+#include <raft/random/rng_state.hpp>
+
+#include <thrust/for_each.h>
+
+#include <iostream>
+#include <string>
+
+void initialize_mpi_and_set_device(int argc, char** argv)
+{
+  RAFT_MPI_TRY(MPI_Init(&argc, &argv));
+
+  int comm_rank{};
+  RAFT_MPI_TRY(MPI_Comm_rank(MPI_COMM_WORLD, &comm_rank));
+
+  int comm_size{};
+  RAFT_MPI_TRY(MPI_Comm_size(MPI_COMM_WORLD, &comm_size));
+
+  int num_gpus_per_node{};
+  RAFT_CUDA_TRY(cudaGetDeviceCount(&num_gpus_per_node));
+  RAFT_CUDA_TRY(cudaSetDevice(comm_rank % num_gpus_per_node));
+}
+
+std::unique_ptr<raft::handle_t> initialize_mg_handle(std::string const& allocation_mode = "cuda")
+{
+  int comm_rank{};
+  RAFT_MPI_TRY(MPI_Comm_rank(MPI_COMM_WORLD, &comm_rank));
+
+  std::set<std::string> possible_allocation_modes = {"cuda", "pool", "binning", "managed"};
+
+  if (possible_allocation_modes.find(allocation_mode) == possible_allocation_modes.end()) {
+    if (!comm_rank) {
+      std::cout << "'" << allocation_mode
+                << "' is not a valid allocation mode. It must be one of the followings -"
+                << std::endl;
+      std::for_each(possible_allocation_modes.cbegin(),
+                    possible_allocation_modes.cend(),
+                    [](std::string mode) { std::cout << mode << std::endl; });
+    }
+    RAFT_MPI_TRY(MPI_Finalize());
+
+    exit(0);
+  }
+
+  if (!comm_rank) {
```

Comment:
Same here and in this function in other example files as well.

Reply:
changed.